src/frontend/src/components/SubmissionMap/SubmissionInstanceMap.tsx

```diff
@@ -5,9 +5,32 @@ import LayerSwitcherControl from '@/components/MapComponent/OpenLayersComponent/
 import { VectorLayer } from '@/components/MapComponent/OpenLayersComponent/Layers';
 import { defaultStyles } from '@/components/MapComponent/OpenLayersComponent/helpers/styleUtils';
 import LayerSwitchMenu from '../MapComponent/OpenLayersComponent/LayerSwitcher/LayerSwitchMenu';
+import { ClusterLayer } from '@/components/MapComponent/OpenLayersComponent/Layers';
+import { Style, Circle } from 'ol/style';
+import { Stroke } from 'ol/style';
+import { hexToRgba } from '@/components/MapComponent/OpenLayersComponent/helpers/styleUtils';
+import { Fill } from 'ol/style';
+import { geojsonType } from '@/store/types/ISubmissions';
 
 type submissionInstanceMapPropType = {
-  featureGeojson: Record<string, any>;
+  featureGeojson: { vectorLayerGeojson: geojsonType; clusterLayerGeojson: geojsonType };
+};
+
+const getIndividualStyle = (featureProperty) => {
+  const style = new Style({
```

Comment:
This PR is great!

The only thing I wanted to add was the possibility of displaying polygons and linestrings too 😄

Could the style be adapted if the geom is a line or polygon?

Reply:
Yeah it works completely fine. 
It seems that for the features in java rosa format odk converts polygon to LineString as well so though its line or polygon it not a problem. 
So I tested with many type of geometries and all of those were visible. 
And hope it works in future too on further testing

Reply:
Amazing! Great work testing 👍

I saw the different javarosa geom extraction, but at first glance thought the only style was circle - my bad!